alarm_active_table.cpp

```diff
@@ -0,0 +1,423 @@
+static oid alarm_active_state_oid[] = { ALARM_ACTIVE_STATE_OID };
+static oid alarm_clear_state_oid[] = { ALARM_CLEAR_STATE_OID };
+
+static oid entry_column_oid[] = { ENTRY_COLUMN_OID };
+static size_t entry_column_oid_len = OID_LENGTH(entry_column_oid);
+
+std::map<unsigned long, alarmActiveTable_context*> pointers;
+std::string loc_ip;
+
+/************************************************************
+ *
+ *  Initialize the alarmActiveTable table by defining its contents
+ *  and how it's structured
+ */
+int init_alarmActiveTable(std::string ip)
+{
+  loc_ip = ip;
+  netsnmp_table_registration_info *table_info;
+
+  if (my_handler)
+  {
+    TRC_ERROR("initialize_table_alarmActiveTable called again");
+    return SNMP_ERR_NOERROR;
+  }
+
+  memset(&cb, 0x00, sizeof(cb));
+
+  /** create the table structure itself */
+  table_info = SNMP_MALLOC_TYPEDEF(netsnmp_table_registration_info);
+
+  my_handler = netsnmp_create_handler_registration("alarmActiveTable",
+                                                   netsnmp_table_array_helper_handler,
+                                                   alarmActiveTable_oid,
+                                                   alarmActiveTable_oid_len,
+                                                   HANDLER_CAN_RONLY);
+            
+  if (!my_handler || !table_info)
+  {
+    TRC_ERROR("malloc failed: initialize_table_alarmActiveTable");
+    return SNMP_ERR_GENERR;
+  }
+
+  /*
+   * Setting up the table's definition
+   */
+
+  /** index: alarmListName */
+  netsnmp_table_helper_add_index(table_info, ASN_OCTET_STR);
+  /** index: alarmActiveDateAndTime */
+  netsnmp_table_helper_add_index(table_info, ASN_OCTET_STR);
+  /** index: alarmActiveIndex */
+  netsnmp_table_helper_add_index(table_info, ASN_UNSIGNED);
+
+  table_info->min_column = alarmActiveTable_COL_MIN;
+  table_info->max_column = alarmActiveTable_COL_MAX;
+
+  /*
+   * registering the table with the master agent
+   */
+  cb.get_value = alarmActiveTable_get_value;
+  cb.container = netsnmp_container_find("alarmActiveTable_primary:"
+                                        "alarmActiveTable:"
+                                        "table_container");
+  cb.can_set = 0;
+
+  DEBUGMSGTL(("initialize_table_alarmActiveTable", "Registering as table array\n"));
+
+  netsnmp_table_container_register(my_handler, table_info, &cb, cb.container, 1);
+
+  return SNMP_ERR_NOERROR;
+}
+
+/************************************************************
+ *
+ *  This routine is called for get requests to copy the data
+ *  from the context to the varbind for the request.
+ */
+int alarmActiveTable_get_value(netsnmp_request_info* request,
+                              netsnmp_index* item,
+                              netsnmp_table_request_info* table_info)
+{
+  netsnmp_variable_list* var = request->requestvb;
+  alarmActiveTable_context* context = (alarmActiveTable_context*) item;
+  switch(table_info->colnum)
```

Comment:
Normally we'd have a space here - `switch (table_info->colnum)` - like for if statements.